src/API/Authentication.php

```diff
@@ -170,46 +217,57 @@ public function sms_passwordless_start($phone_number)
 
     /**
-     * @param $accessToken
+     * @param string $accessToken
      *
-     * @return array|string
+     * @return array
      */
-    public function userinfo($accessToken)
+    public function userInfo($accessToken)
     {
         $response = $this->httpClient->get('/userinfo', ['Authorization' => 'Bearer '.$accessToken]);
 
         return ResponseMediator::getContent($response);
     }
 
-    public function impersonate($access_token, $user_id, $protocol, $impersonator_id, $client_id, $additionalParameters = [])
+    /**
+     * @param string $accessToken
+     * @param string $userId
+     * @param string $protocol
+     * @param string $impersonatorId
+     * @param string $clientId
+     * @param array $additionalParameters
+     *
+     * @return array
+     */
+    public function impersonate($accessToken, $userId, $protocol, $impersonatorId, $clientId, array $additionalParameters = [])
     {
         $data = [
             'protocol'             => $protocol,
-            'impersonator_id'      => $impersonator_id,
-            'client_id'            => $client_id,
+            'impersonator_id'      => $impersonatorId,
+            'client_id'            => $clientId,
             'additionalParameters' => $additionalParameters,
         ];
 
-        $response = $this->httpClient->post(sprintf('/users/%s/impersonate', $user_id), ['Authorization' => 'Bearer '.$access_token], json_encode($data));
+        $response = $this->httpClient->post(sprintf('/users/%s/impersonate', $userId), ['Authorization' => 'Bearer '.$accessToken], json_encode($data));
 
         return ResponseMediator::getContent($response);
     }
 
     /**
      * Makes a call to the `oauth/token` endpoint.
      *
-     * @method oauth_token
+     * @param string $grantType
+     * @param array $options {
+     *      @var string $client_id     Optional. Will use $this->clientId as default
+     *      @var string $client_secret Optional. Only if grant type: client_credentials. Will use $this->clientId as default
+     *      @var string $username      Optional. Only if grant type: password/password-realm
+     *      @var string $password      Optional. Only if grant type: password/password-realm
+     *      @var string $scope         Optional.
+     *      @var string $audience      Optional.
+     * }
      *
-     * @param {Object} options:
-     * @param {Object} options.grantType
-     * @param {Object} options.client_id
-     * @param {Object} options.client_secret [optional] Only if grant type: client_credentials
-     * @param {Object} options.username  [optional] Only if grant type: password/password-realm
-     * @param {Object} options.password  [optional] Only if grant type: password/password-realm
-     * @param {Object} options.scope     [optional]
-     * @param {Object} options.audience  [optional]
+     * @return array
      */
-    public function oauth_token($options = [])
+    public function oauthToken($grantType, array $options)
```

Comment:
Should this become a private method, it seems to mostly be a wrapper now @glena ?